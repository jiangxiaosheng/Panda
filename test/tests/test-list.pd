func main() {
	var l: int[]
	l = [1, 2, 3]
    l

    var sl: string[]
	sl = ["1", "2", "3"]
    sl

    var fl: float[]
	fl = [1.0, 2.0, 3.0]
    fl

    var bl: boolean[]
	bl = [true, false]
    bl

}