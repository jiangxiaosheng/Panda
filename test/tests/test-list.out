Sementically checked program: 


void main() {
var l: list of int(list of int : [])
(list of int : l = (list of int : [(int : 1), (int : 2), (int : 3)]))
(list of int : l)
var sl: list of string(list of string : [])
(list of string : sl = (list of string : [(string : 1), (string : 2), (string : 3)]))
(list of string : sl)
var fl: list of float(list of float : [])
(list of float : fl = (list of float : [(float : 1.), (float : 2.), (float : 3.)]))
(list of float : fl)
var bl: list of bool(list of bool : [])
(list of bool : bl = (list of bool : [(bool : true), (bool : false)]))
(list of bool : bl)
}

